whipper/command/mblookup.py

```diff
@@ -12,35 +12,52 @@ class MBLookup(BaseCommand):
 
     def add_arguments(self):
         self.parser.add_argument(
-            'mbdiscid', action='store', help="MB disc id to look up"
+            'id', action='store', help="MB disc id or release id to look up"
         )
 
+    def _printMetadata(self, md):
+        """
+        Print out metadata received in a sensible way
+
+        :param md: MusicBrainz's metadata about the disc
+        :type md: `DiscMetadata`
+        rtype: None
+        """
+        print('    Artist: %s' % md.artist.encode('utf-8'))
+        print('    Title:  %s' % md.title.encode('utf-8'))
+        print('    Type:   %s' % str(md.releaseType).encode('utf-8'))
+        print('    URL:    %s' % md.url)
+        print('    Tracks: %d' % len(md.tracks))
+        if md.catalogNumber:
+            print('    Cat no: %s' % md.catalogNumber)
+        if md.barcode:
+            print('    Barcode: %s' % md.barcode)
+
+            for j, track in enumerate(md.tracks):
+                print('      Track %2d: %s - %s' % (
+                    j + 1, track.artist.encode('utf-8'),
+                    track.title.encode('utf-8')
+                ))
+        return None
+
     def do(self):
         try:
-            discId = str(self.options.mbdiscid)
+            id = str(self.options.id)
         except IndexError:
-            print('Please specify a MusicBrainz disc id.')
+            print('Please specify a MusicBrainz disc id or release id.')
             return 3
 
-        metadatas = musicbrainz(discId)
-
-        print('%d releases' % len(metadatas))
-        for i, md in enumerate(metadatas):
-            print('- Release %d:' % (i + 1, ))
-            print('    Artist: %s' % md.artist.encode('utf-8'))
-            print('    Title:  %s' % md.title.encode('utf-8'))
-            print('    Type:   %s' % str(md.releaseType).encode('utf-8'))  # noqa: E501
-            print('    URL:    %s' % md.url)
-            print('    Tracks: %d' % len(md.tracks))
-            if md.catalogNumber:
-                print('    Cat no: %s' % md.catalogNumber)
-            if md.barcode:
-                print('    Barcode: %s' % md.barcode)
-
-                for j, track in enumerate(md.tracks):
-                    print('      Track %2d: %s - %s' % (
-                        j + 1, track.artist.encode('utf-8'),
-                        track.title.encode('utf-8')
-                    ))
-
+        # release id is is 36 characters long and disc id is 28 characters long
+        # see https://musicbrainz.org/doc/MusicBrainz_Identifier
+        if len(id) == 36:
```

Comment:
Maybe you can use a RegEX instead of a length check for the Release ID:

```regex
[\dA-Fa-f]{8}-(?:[\dA-Fa-f]{4}-){3}[\dA-Fa-f]{12}
```

Reply:
My knowledge about RegEx is limited so i could only think of checking length.

I will implement it now.

Reply:
Sorry, forgot to add anchors. Revised RegEX:

```regex
^[\dA-Fa-f]{8}-(?:[\dA-Fa-f]{4}-){3}[\dA-Fa-f]{12}$
```

Reply:
No problem, i fixed this too.